src/dns_server.rs

```diff
@@ -0,0 +1,187 @@
+use log::{info};
+use tokio;
+use tokio::net::{UdpSocket, ToSocketAddrs};
+use tokio::sync::Mutex;
+use trust_dns_proto::op::{Message};
+use trust_dns_proto::rr::{Record, RecordType, DNSClass, RData, Name};
+use std::net::{Ipv4Addr};
+use trust_dns_proto::serialize::binary::BinEncodable;
+use std::sync::Arc;
+
+use crate::mapping::Ipv4Mapping;
+
+
+const BASE_IPV4_ADDR: u32 = 0x0D000000;
+const FIRST_IPV4_ADDR: u32 = BASE_IPV4_ADDR + 1;
+
+#[derive(Debug)]
+pub struct DnsServer {
+    dns_server: UdpSocket,
+    answer_index: u32,
+    ipv4_mapping: Arc<Mutex<Ipv4Mapping>>,
```

Comment:
I presonally like to  type complex types to make the code more readable
 fe type Ipv4MutexedMapping = Arc<Mtex<blablabalba
your choice

Reply:
I prefer to leave it that way so anyone can understand it's a mutex wrapper around `Ipv4Mapping`

Reply:
Since it's used once it's cool. Yet if it will be used oftenly I will instit on typing it